lib/iris/fileformats/netcdf.py

```diff
@@ -1924,36 +1937,55 @@ def set_packing_ncattrs(cfvar):
             self._dataset.file_format in ('NETCDF3_CLASSIC',
```

Comment:
Do we still need the ``has_lazy_data`` here? It'd be nice to remove.

Reply:
It determines which `store` method gets defined, although I could move the `if` into that method if you like.